pageserver/src/tenant/tasks.rs

```diff
@@ -363,7 +363,7 @@ async fn gc_loop(tenant: Arc<Tenant>, cancel: CancellationToken) {
                 first = false;
 
                 let delays = async {
-                    delay_by_lease_length(tenant.get_lsn_lease_length(), &cancel).await?;
+                    tenant.gc_block.block_for(tenant.get_lsn_lease_length(), &cancel).await?;
```

Comment:
Arguably not necessary to acquire the lock and only need to wait for lease duration.

Reply:
Oops, the `tokio::time::timeout(tenant.lsn_lease_length(), cancel.cancelled()).await` would of course be the smallest amount :)